src/utils/gitHubUtils.ts

```diff
@@ -167,3 +171,34 @@ export function isUser(orgData: UsersGetAuthenticatedResponseData | OrgForAuthen
     // if there's no orgData, just assume that it's a user (but this shouldn't happen)
     return !!orgData && 'type' in orgData && orgData.type === 'User';
 }
+
+export async function getGitHubTree(repositoryUrl: string, branch: string): Promise<GitTreeData[]> {
+    const octokitClient: Octokit = await createOctokitClient();
+    const { owner, name } = getRepoFullname(repositoryUrl);
+    const branchRes: OctokitResponse<ReposGetBranchResponseData> = await octokitClient.repos.getBranch({ owner, repo: name, branch });
+    const getTreeRes: OctokitResponse<GitGetTreeResponseData> = await octokitClient.git.getTree({ owner, repo: name, tree_sha: branchRes.data.commit.sha, recursive: 'true' });
+
+    // sort descending by the depth of subfolder
+    return getTreeRes.data.tree.filter(file => file.type === 'tree').sort((f1, f2) => {
+        // tslint:disable-next-line: strict-boolean-expressions
+        function getFolderDepth(path: string): number { return (path.match(/\//g) || []).length; }
+        return getFolderDepth(f1.path) - getFolderDepth(f2.path);
+    });
+}
+
+export async function getGitTreeQuickPicks(wizardContext: IStaticWebAppWizardContext, isSkippable?: boolean): Promise<IAzureQuickPickItem<string>[]> {
+    if (!wizardContext.gitTreeData) {
```

Comment:
You can remove `gitTreeData` from the wizard context. No matter how many times you call `await` on the same task, it's still only executed once, so there's no performance benefit from caching the result on the wizardContext